src/editor/QueryField.tsx

```diff
@@ -256,7 +283,7 @@ class QueryField extends React.Component<any, any> {
 
   handleClickMenu = (item, editor: CoreEditor) => {
     // Manually triggering change
-    const change = this.applyTypeahead(editor, item);
+    const change = this.applyTypeahead(editor, new Suggestion(item));
```

Comment:
`onChange` takes an editor, should be called that. Also see comment above about `applyTypeahead` not doing much.

Reply:
Yeah, I agree. Let me see if I can clean that one up a bit.

Reply:
Now that I've looked it through, apply typeahead handles the suggestion and returns the editor. Applytypeahead is overridden, and just needs something as the parent class.

Reply:
https://github.com/grafana/influxdb-flux-datasource/issues/59